### ISSUE TITLE
S3TaskHandler continuously returns "*** Falling back to local log" even if log_pos is provided when log not in s3

### ISSUE THREAD

**Issue by pdebelak** [apache/airflow#29393]
### Apache Airflow Provider(s)

amazon

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon==7.1.0

### Apache Airflow version

2.5.1

### Operating System

Ubuntu 18.04

### Deployment

Official Apache Airflow Helm Chart

### What happened

When looking at logs in the UI for a running task when using remote s3 logging, the logs for the task are only uploaded to s3 after the task has completed. The the `S3TaskHandler` falls back to the local logs stored on the worker in that case (by falling back to the `FileTaskHandler` behavior) and prepends the line `*** Falling back to local log` to those logs.

This is mostly fine, but for the new log streaming behavior, this means that `*** Falling back to local log` is returned from `/get_logs_with_metadata` on each call, even if there are no new logs.

### What you think should happen instead

I'd expect the falling back message only to be included in calls with no `log_pos` in the metadata or with a `log_pos` of `0`.

### How to reproduce

Start a task with `logging.remote_logging` set to `True` and `logging.remote_base_log_folder` set to `s3://something` and watch the logs while the task is running. You'll see `*** Falling back to local log` printed every few seconds.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by pdebelak** [comment]
There's some comments in `S3TaskHandler._read` about airflow 2.6 that seem to indicate that this fallback will work differently starting in airflow 2.6, but I'm not too sure what is changing in that version.

**Comment by ccwillia** [comment]
I am also seeing the `*** Falling back to local log` message get printed every few seconds in version 2.5 while using s3 logging.  Has anyone found a way to prevent this from being printed?

**Comment by PodbiAtGS** [comment]
Same here. Any ideas?

**Comment by goodseog** [comment]
Same here :)

**Comment by potiuk** [comment]
cc: @dstandish - maybe you can comment on how it changes in 2.6 (and whether the problem is going to be fixed there).

**Comment by pdebelak** [comment]
It seems like in the code as currently written since `FileTaskHandler` implements `_read_remote_logs` this function won't be called at all (see https://github.com/apache/airflow/blob/main/airflow/providers/amazon/aws/log/s3_task_handler.py#L123-L125). But in the current airflow release `FileTaskHandler` doesn't have that method which is why this is called. As a result, I'm not even sure it's possible to write a test for this behavior without mocking out an old version of the superclass.

**Comment by dstandish** [comment]
Hi y'all, thanks for the issue reports :) 

I did not spend nearly as much time live testing the legacy logs view as I did with the grid logs view.  Is there some reason y'all prefer the legacy logs view?  Just curious.

I will try and look at this but honestly, I have a _lot_ going on right now and if one of you who have looked into the issue can contribute a fix that would be really welcome.  I would be delighted to review any PR.
